### PR TITLE
Better error message for unpacking-non-sequence and unbalanced-tuple-unpacking

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Necessary to be able to do https://github.com/PyCQA/pylint/pull/7750 without thinking about it.

### CODE REVIEW

**Review by clavedeluna — APPROVED** [review]

File: pylint/checkers/variables.py [hunk 1]
```diff
@@ -2673,18 +2672,18 @@ def _check_unpacking(
                     "unbalanced-tuple-unpacking",
                     node=node,
                     args=(
-                        _get_unpacking_extra_info(node, inferred),
+                        details,
                         len(targets),
+                        "s" if len(targets) > 1 else "",
```
Comment (mbyrnepr2) [inline pylint/checkers/variables.py:2677]:
```suggestion
                        "" if len(targets) == 1 else "s",
```

File: pylint/checkers/variables.py [hunk 2]
```diff
@@ -2673,18 +2672,18 @@ def _check_unpacking(
...
                         len(values),
+                        "s" if len(values) > 1 else "",
```
Comment (mbyrnepr2) [inline pylint/checkers/variables.py:2679]:
```suggestion
                        "" if len(values) == 1 else "s",
```

**Review by mbyrnepr2 — COMMENTED** [review]
Looks great. I like the care for usability. Small change required I think!
The primer gives: `right side has 0 value`

File: pylint/checkers/variables.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply pylint/checkers/variables.py:2677]: Damn TIL ! a search for "0 values or 0 value" bring nothing but it seems that apple is the universal example. https://www.quora.com/Is-zero-plural-You-have-one-apple-I-have-zero-appleS

**Review by Pierre-Sassoulas — COMMENTED** [review]
Thank you for the grammar lesson @mbyrnepr2 😉

**Comment by mbyrnepr2** [comment]
> Thank you for the grammar lesson @mbyrnepr2 😉

Zero problems 😅

File: pylint/checkers/variables.py (hunk 1 above)
Comment (mbyrnepr2) [reply pylint/checkers/variables.py]: 😁🍎

**Review by mbyrnepr2 — APPROVED** [review]

**Review by clavedeluna — COMMENTED** [review]
nice!

File: pylint/checkers/variables.py [hunk 3]
```diff
@@ -2671,6 +2669,7 @@ def _check_unpacking(
 
         # Attempt to check unpacking is properly balanced
         values = self._nodes_to_unpack(inferred)
+        details = _get_unpacking_extra_info(node, inferred)
```
Comment (clavedeluna) [inline pylint/checkers/variables.py:2672]: this prob should've been below the if statement in ln 2675 but no big deal, I'll move it in the next PR

Comment (clavedeluna) [reply pylint/checkers/variables.py:2672]: nvm, I see it's used below!

Comment (Pierre-Sassoulas) [reply pylint/checkers/variables.py:2672]: No I thought of it too (and then forgot). Performance wise this is pretty bad, we'd better duplicate the code so we only create the details when something is going to be raised.

**Pull request merged**